plugins/processors/ifname/ifname.go

```diff
@@ -241,60 +241,61 @@ func (d *IfName) Stop() error {
 func (d *IfName) getMap(agent string) (entry nameMap, age time.Duration, err error) {
 	var sig chan struct{}
 
+	d.lock.Lock()
+
 	// Check cache
-	d.rwLock.RLock()
 	m, ok, age := d.cache.Get(agent)
-	d.rwLock.RUnlock()
 	if ok {
+		d.lock.Unlock()
 		return m, age, nil
 	}
 
-	// Is this the first request for this agent?
-	d.rwLock.Lock()
+	// cache miss.  Is this the first request for this agent?
 	sig, found := d.sigs[agent]
 	if !found {
+		// This is the first request.  Make signal for subsequent requests to wait on
 		s := make(chan struct{})
 		d.sigs[agent] = s
 		sig = s
 	}
-	d.rwLock.Unlock()
+
+	d.lock.Unlock()
 
 	if found {
 		// This is not the first request.  Wait for first to finish.
 		<-sig
+
 		// Check cache again
-		d.rwLock.RLock()
+		d.lock.Lock()
 		m, ok, age := d.cache.Get(agent)
-		d.rwLock.RUnlock()
+		d.lock.Unlock()
 		if ok {
 			return m, age, nil
 		}
 		return nil, 0, fmt.Errorf("getting remote table from cache")
 	}
 
 	// The cache missed and this is the first request for this
-	// agent.
-
-	// Make the SNMP request
+	// agent. Make the SNMP request
 	m, err = d.getMapRemote(agent)
+
+	d.lock.Lock()
 	if err != nil {
-		//failure.  signal without saving to cache
-		d.rwLock.Lock()
+		//snmp failure.  signal without saving to cache
 		close(sig)
 		delete(d.sigs, agent)
-		d.rwLock.Unlock()
 
+		d.lock.Unlock()
 		return nil, 0, fmt.Errorf("getting remote table: %w", err)
 	}
 
-	// Cache it, then signal any other waiting requests for this agent
-	// and clean up
-	d.rwLock.Lock()
+	// snmp success.  Cache response, then signal any other waiting
```

Comment:
nit: any reason for having the two spaces after the period?

Reply:
When I learned to type that's what they taught in school. I'm aware that single space is more common now but it's a habit.

See https://en.wikipedia.org/wiki/Sentence_spacing